controllers/templatesync/template_sync.go

```diff
@@ -1222,7 +1222,11 @@ func overrideRemediationAction(instance *policiesv1.Policy, tObjectUnstructured
 		if spec, ok := tObjectUnstructured.Object["spec"]; ok {
 			specObject, ok := spec.(map[string]interface{})
 			if ok {
-				specObject["remediationAction"] = string(instance.Spec.RemediationAction)
+				if strings.EqualFold(specObject["remediationAction"].(string), "informonly") {
```

Comment:
Because the policy template inside a Policy isn't strictly typed (and it can't be strictly typed because we don't know if it will be a ConfigurationPolicy, or a Gatekeeper resource, or something else entirely), we need to be careful every time we look up a field and expect it to be a certain type. In this case, if the field doesn't exist, or isn't a string, I think this code will panic and crash the controller. So I'd rather see the lookup and the string type assertion be separated, and individually checked - it's a hassle, but it helps in case there are typos introduced by the user.